qiskit/utils/optionals.py

```diff
@@ -142,6 +142,10 @@
       - `Z3 <https://github.com/Z3Prover/z3>`__ is a theorem prover, used in the
         :class:`.CrosstalkAdaptiveSchedule` and :class:`.HoareOptimizer` transpiler passes.
 
+    * - .. py:data:: HAS_CONSTRAINT
+      - `python-constraint <https://github.com/python-constraint/python-constraint>__ is a
+        constraint satisfaction problem solver, used in the :class:`~.CSPLayout` transpiler pass.
+
```

Comment:
I think it's worth keeping these in alphabetical order in the documentation (not necessarily the code), within the sections to make searching/reading easier. So this object should go to the top of this section.

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/7733/commits/9b6915d0737c29a7cd72dc046d78223f0e09ed36